modules/programs/tmux.nix

```diff
@@ -0,0 +1,203 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.tmux;
+
+  mkChildPackageType = p: (
+   let
+     ptokens = splitString p.name "";
+     ptitle = toUpper (head ptokens) + concatStrings (tail ptokens);
+   in types.submodule {
+    options = {
+      enable = mkEnableOption "${ptitle}";
+
+      package = mkOption {
+        type = types.package;
+        default = p;
+        defaultText = "pkgs.${p.name}";
+        description = "Also install ${ptitle}.";
+      };
+    };
+  });
+
+  mkTmuxOption = name: opts: (types.submodule {
+    options = mkMerge [{
+      # FIXME: should be renamed.
+      specifier = mkOption {
+        type = types.string;
+        description = "Type of ${name}.";
+        default = "";
+        example = "\"-key\" becomes \"${name}-key\".";
+      };
+      flags = mkOption {
+        type = nullOr types.string;
+        description = "Flags applied to the set variable. Implies '-' character.";
+        default = null;
+        example = "\"ag\" becomes \"-ag\".";
+      };
+      key = mkOption {
+        type = types.string;
+        description = "Key to set.";
+        example = "mode-keys";
+      };
+    }
+    opts ];
+  });
+
+
+  bindOption = mkTmuxOption "bind" {
+    values = mkOption {
+      type = type.listOf types.string;
+      description = "variable list of arguments to bind to key.";
+    };
+  };
+
+  setOption = mkTmuxOption "set" {
+    value = mkOption {
+      type = types.string;
+      description = "Value to set key to.";
+      example = "vi";
+    };
+  };
+
+  unbindOption = mkTmuxOption "unbind" { };
+
+  tmuxinatorModule = mkChildPackageType pkgs.tmuxinator;
+  tmuxpModule = mkChildPackageType pkgs.tmuxp;
+  hasExtraConfig = x: hasAttr "extraConfig" x && x.extraConfig != "";
+  pluginName = p: if types.package.check p then p.name else p.plugin.name;
+
+  pluginModule = types.submodule {
+    options = {
+      plugin = mkOption {
+        type = types.package;
+        description = "Path of the configuration file to include.";
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        description = "Additional configuration for the associated plugin.";
+        default = "";
+      };
+    };
+  };
+in
+{
+  options = {
+
+    programs.tmux = {
+      enable = mkEnableOption "Tmux";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.tmux;
+        defaultText = "pkgs.tmux";
+        example = literalExample "pkgs.tmux";
+        description = "tmux";
+      };
+
+      sensibleOnTop = mkOption {
+        type = types.bool;
+        default = false;
+        description = "run the sensible plugin at the top of your .conf (to overwrite configs)";
```

Comment:
this phrasing could actually be a little better.

Reply:
Would it be possible to simply always including the sensible plugin? Is it something that everybody uses?

Reply:
I don't think so -- I wasn't even aware of it when I first got started with tmux plugins. It's _probably_ best to do this, but I really can't give you a good answer. Reflecting back on this, however, I think this should default to `true`